libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java

```diff
@@ -883,17 +882,31 @@ public void onClick(DialogInterface dialog, int id) {
 
                 builder.setNegativeButton(getString(R.string.cancel), new DialogInterface.OnClickListener() {
                     public void onClick(DialogInterface dialog, int id) {
+                        try {
+                            // only retry if id is valid
+                            Integer.parseInt(localMediaId);
+                            mEditorFragmentListener.onMediaRetryClicked(localMediaId);
+                        } catch (NumberFormatException e) {
+                            // invalid id
+                        }
                         dialog.dismiss();
                     }
                 });
 
+                // stop the media upload while the dialog is being shown
+                // otherwise the upload might finish and we won't find the element for removal
```

Comment:
You're right, it is drastic and it's true the upload is restarted. I just couldn't figure out the a better solution. But silently failing with a toast is a reasonable compromise so I'll go ahead with that. Thanks for the tip 👍.